libexec/tgenv-install

```diff
@@ -60,19 +61,38 @@ tarball_name="terragrunt_${os}";
 
 info "Installing Terragrunt v${version}"
 
+function install_tg_version() {
+  local http_code
+  http_code=$(curlw -# --location-trusted -w "%{http_code}" -o "${dst_path}/terragrunt" "${version_url}/${tarball_name}")
+  local curl_exit_code=$?
+
+  if [[ $http_code =~ ^4 ]]; then
+    echo "Got $http_code http code, TG version is invalid"
+    return 2 # do not retry
+  elif [[ $http_code =~ ^5 ]]; then
+    echo "Got $http_code http code"
+    return 1 # retry
+  elif [[ $curl_exit_code -ne 0 ]]; then
+    return 1 # retry
+  else
+    return 0 # success
+  fi
+}
+
+
 mkdir -p "${dst_path}" || error_and_die "Failed to make directory ${dst_path}"
 
 info "Downloading release tarball from ${version_url}/${tarball_name}"
-curlw -# --location-trusted -f -o "${dst_path}/terragrunt" "${version_url}/${tarball_name}"
-curl_status=$?
-if [ "$curl_status" -eq 0 ]; then
+retry -tries 5 -sleep 5 -stop-on-exit-code 2 install_tg_version || error_and_die "Tarball download failed"
```

Comment:
won't this `error_and_die` here will just make the code stop running, and we won't get to the cleanup `else` block (lines 96-97) at all?

Reply:
True, ill fix that wnd probbaly will move 
```
# make sure versions folder does not exist, so we can retry installation
  rm -rf "${dst_path}"
```

to `install_tg_version`

Reply:
@ronnyorot [fix folder cleanup after failed install](https://github.com/env0/tgenv/pull/11/commits/0cb0f396cb358c7f16fe023c722f718d7d7ee495)

great catch!